src/renderer/components/CodeEditor/SqlCodeEditor.tsx

```diff
@@ -6,20 +6,94 @@ import {
   acceptCompletion,
   completionStatus,
   startCompletion,
+  autocompletion,
+  CompletionContext,
+  CompletionResult,
 } from '@codemirror/autocomplete';
 import { defaultKeymap, insertTab } from '@codemirror/commands';
 import { keymap } from '@codemirror/view';
-import { SQLConfig, sql, MySQL } from '@codemirror/lang-sql';
-import { Ref, forwardRef } from 'react';
+import { Ref, forwardRef, useCallback } from 'react';
 import useCodeEditorTheme from './useCodeEditorTheme';
+import type { EnumSchema } from 'renderer/screens/DatabaseScreen/QueryWindow';
+import { SyntaxNode } from '@lezer/common';
+import {
+  SQLConfig,
+  sql,
+  MySQL,
+  genericCompletion,
+  keywordCompletionSource,
+  schemaCompletionSource,
+} from 'query-master-lang-sql';
 
 const SqlCodeEditor = forwardRef(function SqlCodeEditor(
-  props: ReactCodeMirrorProps & { schema: SQLConfig['schema'] },
+  props: ReactCodeMirrorProps & {
+    schema: SQLConfig['schema'];
+    enumSchema: EnumSchema;
+  },
   ref: Ref<ReactCodeMirrorRef>
 ) {
-  const { schema, ...codeMirrorProps } = props;
+  const { schema, enumSchema, ...codeMirrorProps } = props;
   const theme = useCodeEditorTheme();
 
+  const enumCompletion = useCallback(
+    (context: CompletionContext, tree: SyntaxNode): CompletionResult | null => {
+      // dont run if there is no enumSchema
+      if (enumSchema.length === 0) return null;
+
+      const isEqualOperator = tree.prevSibling;
+
+      // check if it's an operator and id is 15
+      if (
+        !(
+          isEqualOperator?.type.name === 'Operator' &&
+          isEqualOperator?.type.id === 15
+        )
+      )
+        return null;
+
+      // get the column name before the equal operator node
+      const tableAndColumnNode = isEqualOperator?.prevSibling;
+      if (!tableAndColumnNode) return null;
+
+      // get the text from tableAndColumnNode
+      // first convert context.state.doc to string
+      // then slice from node.from to node.to
+      const tableAndColumnText = context.state.doc
+        .toString()
+        .slice(tableAndColumnNode.from, tableAndColumnNode.to);
+
+      // split the text to table and column
+      const [table, column] = tableAndColumnText.split('.');
+      // only check for table since column is possibly a keyword
+      if (!table) return null;
+
+      const enumValues = enumSchema.find((tempEnum) => {
+        if (column) {
+          // normally column will be enum
+          return tempEnum.column === column.replaceAll('`', '');
```

Comment:
It would be expensive to repeating replaceAll which can just do it one time at the top.

Reply:
How did I not think of this? 😱

Reply:
I have moved replaceAll higher up